hs_core/models.py

```diff
@@ -961,6 +961,36 @@ def __unicode__(self):
         return self.value
 
 
+class FundingAgency(AbstractMetaDataElement):
+    term = 'FundingAgency'
+    agency_name = models.TextField(null=False)
+    award_title = models.TextField(null=True, blank=True)
+    award_number = models.TextField(null=True, blank=True)
+    agency_url = models.URLField(null=True, blank=True)
+
+    class Meta:
+        unique_together = ("content_type", "object_id", "award_number", "agency_name")
```

Comment:
@pkdash @horsburgh Thanks for the clarification. I guess my question is whether we need to make award number required as well which will ensure uniqueness of award using agency name and award_number. Currently, award_number is allowed to be null, although unlikely we could have more than one funding agencies with same name and blank award_number, which cannot be differentiated. I guess we don't want to enforce award_number as a required field since some award may not have an award_number. @pkdash I am not sure whether django database validation will complain if two awards have same funding agency names and blank award_numbers with current unique_together definition.